core/Result.carp

```diff
@@ -106,6 +108,21 @@ It is the inverse of [`success?`](#success?).")
         (match @b
           (Success _) false
           (Error y) (= x y))))
+
+  (doc ref= "reference equality is defined as the equality of both the
+constructor and the contained value, i.e. `(Success [1])` and
+`(Success [1])` are the same, but `(Success [1])` and `(Success [2])` are
+not.")
+  (defn = [a b]
+    (match @a
+      (Success x)
+        (match @b
+          (Error _) false
+          (Success y) (= &x &y))
+      (Error x)
+        (match @b
+          (Success _) false
+          (Error y) (= &x &y))))
```

Comment:
Should these and other ref comparisons use `ref=` now? And should we implement the interface for other types?

Reply:
I guess this is the question I was asking myself while doing this (and the world at large when writing the PR description). The problem is that equality is on references by default for strings and it would probably be a big breaking change to change this to `ref=`, but I think that would probably be the best way to go in the future? Implementing `=` twice in different modules for the same type has always been kind of annoying for me.